testutil/network.go

```diff
@@ -8,11 +8,40 @@ package testutil
 
 import (
 	"math/rand"
+	"net"
+	"strconv"
 	"time"
+
+	"github.com/pkg/errors"
 )
 
+var ErrPortOpened = errors.New("tcp port is opened")
+
+func CheckPortIsOpen(port int) error {
+	timeout := time.Millisecond * 10
+	conn, err := net.DialTimeout("tcp", net.JoinHostPort("127.0.0.1", strconv.Itoa(port)), timeout)
+	if err != nil {
+		// port not opened
+		return nil
+	}
+	// port has opened
+	if conn != nil {
+		conn.Close()
+	}
+	return ErrPortOpened
+}
+
 // RandomPort returns a random port number between 30000 and 50000
 func RandomPort() int {
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
-	return r.Intn(20000) + 30000
+	port := r.Intn(20000) + 30000
```

Comment:
i think can use Intn(2000) to start from at most 32000, and give 18000 available ports to try